src/core/lib/layer-manager.js

```diff
@@ -245,19 +260,39 @@ export default class LayerManager {
     }
 
     this.prevLayers = this.layers;
+
     const {error, generatedLayers} = this._updateLayers({
       oldLayers: this.prevLayers,
       newLayers
     });
 
     this.layers = generatedLayers;
+
     // Throw first error found, if any
     if (error) {
       throw error;
     }
     return this;
   }
 
+  // Update layers from last cycle if `setNeedsUpdate()` has been called
+  updateLayers() {
+    // NOTE: For now, even if only some layer has changed, we update all layers
+    // to ensure that layer id maps etc remain consistent even if different
+    // sublayers are rendered
+    const reason = this.needsUpdate();
+    if (reason) {
+      this.setNeedsRedraw(`updating layers: ${reason}`);
+      // HACK - Call with a copy of lastRenderedLayers to trigger a full update
+      this.setLayers([...this.lastRenderedLayers]);
```

Comment:
Nit: use `.slice()` instead? We used to have problems with perf when spread operator is transpiled.

Reply:
I checked the transpiled code. Very non-obviously babel does a
```
function _toConsumableArray(arr) { if (Array.isArray(arr)) { for (var i = 0, arr2 = Array(arr.length); i < arr.length; i++) { arr2[i] = arr[i]; } return arr2; } else { return Array.from(arr); } }
```
I can believe that code could be a bit slower. However this code path should not be called often, I think it is OK for now.